pandas-stubs/core/series.pyi

```diff
@@ -1025,12 +1032,114 @@ class Series(IndexOpsMixin, NDFrame, Generic[S1]):
         axis: SeriesAxisType | None = ...,
         ignore_index: _bool = ...,
     ) -> Series[S1]: ...
+    @overload
+    # def astype(
+    #     self,
+    #     dtype: Union[
+    #         Literal["int"],
+    #         int,
+    #         pd.Int8Dtype,
+    #         pd.Int16Dtype,
+    #         pd.Int16Dtype,
+    #         pd.Int32Dtype,
+    #         pd.Int64Dtype,
+    #         np.int8,
+    #         np.int16,
+    #         np.int32,
+    #         np.int64,
+    #         np.uint8,
+    #         np.uint16,
+    #         np.uint32,
+    #         np.uint64,
+    #         np.intp,
+    #         np.uintp,
+    #         np.integer,
+    #     ],
     def astype(
         self,
-        dtype: S1 | _str | type[Scalar],
+        dtype: Literal["int"]
+        | type[int]
+        | pd.Int8Dtype
+        | pd.Int16Dtype
+        | pd.Int32Dtype
+        | pd.Int64Dtype
+        | type[np.int8]
+        | type[np.int16]
+        | type[np.int32]
+        | type[np.int64]
+        | type[np.uint8]
+        | type[np.uint16]
+        | type[np.uint32]
+        | type[np.uint64]
+        | type[np.intp]
+        | type[np.uintp]
+        | type[np.integer],
         copy: _bool = ...,
         errors: IgnoreRaise = ...,
-    ) -> Series: ...
+    ) -> Series[int]: ...
+    @overload
+    def astype(
+        self,
+        dtype: type[_str] | pd.StringDtype | Literal["str"],
+        copy: _bool = ...,
+        errors: IgnoreRaise = ...,
+    ) -> Series[str]: ...
+    @overload
+    def astype(
+        self,
+        dtype: type[np.byte] | type[np.ubyte] | bytes,
+        copy: _bool = ...,
+        errors: IgnoreRaise = ...,
+    ) -> Series[bytes]: ...
+    @overload
+    def astype(
+        self,
+        dtype: pd.Float32Dtype
+        | pd.Float64Dtype
+        | type[np.float16]
+        | type[np.float32]
+        | type[np.float64]
+        | type[np.float96]
+        | type[np.float128]
+        | type[np.floating]
+        | type[float]
+        | Literal["float"],
+        copy: _bool = ...,
+        errors: IgnoreRaise = ...,
+    ) -> Series[float]: ...
+    @overload
+    def astype(
+        self,
+        dtype: type[np.complex64]
+        | type[np.complex128]
+        | type[np.complex192]
+        | type[np.complex256]
+        | type[complex]
+        | Literal["complex"],
+        copy: _bool = ...,
+        errors: IgnoreRaise = ...,
+    ) -> Series[complex]: ...
+    @overload
+    def astype(
+        self,
+        dtype: pd.BooleanDtype | type[np.bool_] | type[bool] | Literal["bool"],
```

Comment:
I think either `type[np.bool_]` or `type[bool]` is causing the overlap here, so try removing them one at a time to get the message about sigs 1 and 6 overlapping to go away.